src/libraries/System.Private.Xml/tests/XmlReaderLib/TestFiles.cs

```diff
@@ -291,42 +291,42 @@ protected static void DeleteTestFile(string strFileName)
 
         public static void CreateByteTestFile(string strFileName)
         {
-            Stream s = new MemoryStream();
-            TextWriter tw = new StreamWriter(s);
+            var s = new MemoryStream();
+            using var tw = new StreamWriter(s, encoding:null, bufferSize:-1, leaveOpen:true);
+
             tw.WriteLine("x");
-            tw.Flush();
+
             FilePathUtil.addStream(strFileName, s);
         }
 
         public static void CreateUTF8EncodedTestFile(string strFileName, Encoding encode)
         {
-            Stream strm = new MemoryStream();
-            TextWriter tw = new StreamWriter(strm, encode);
+            var s = new MemoryStream();
+            using var tw = new StreamWriter(s, encode, bufferSize:-1, leaveOpen:true);
```

Comment:
I'm curious about negative `bufferSize`. Per [docs](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=net-6.0) we should be getting `ArgumentOutOfRangeException` for negative values, does that actually mean `don't buffer` or `use default`? Either way we should update the docs accordingly I think